File: easy/py/intersection_of_two_arrays_350.py
#!/usr/bin/env python
# -*- encoding: utf-8 -*-
'''
@File    :   intersection_of_two_arrays_350.py
@Time    :   2020/09/01 15:04:42
@Author  :   RoryXiang 
@Version :   1.0
'''

'''
Given two arrays, write a function to compute their intersection.
'''


# here put the import lib

class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        temp = {}
        for num in nums1:
            temp[num] = temp.get(num, 0) + 1
        k = 0
        for num in nums2:
            if temp.get(num, 0) > 0:
                nums2[k] = num
                k += 1
                temp[num] = temp[num] - 1
        return nums2[:k]

File: easy/go/intersection_of_two_arrays_350.go
package main
import "fmt"


func intersect(nums1 []int, nums2 []int) []int {
    temp := map[int]int{}
    for _,val :=range nums1{
        temp[val] +=1
    }
    k := 0
    for _,v := range nums2{
        if temp[v] > 0{
            temp[v] -= 1
            
            nums2[k] = v
            k++
        }
    }
    return nums2[0:k]
}

func main (){
	var t1 = []int{1,2,3,4,2}
	var t2 = []int{2,9,2}
	var a = intersect(t1,t2)
	fmt.Println(a)
	
}